### PR TITLE
Add hex Format string to lua string.format

### PR DESCRIPTION
**Pull request by eacmen**
This allows developers to use the hex format string %x or %X in lua scripts to print hexadecimal PIDs versus float or integer values. 

`make test` passed.

### CODE REVIEW

File: lib/lua/src/lstrlib.c [hunk 1]
```diff
@@ -854,6 +854,12 @@ static int str_format (lua_State *L)
                 modp_itoa10((LUA_INTFRM_T)luaL_checknumber(L, arg),buff);
                 break;
             }
+            case 'x':
+            case 'X': {
+                addintlen(form);
+                modp_itoaX((LUA_INTFRM_T)luaL_checknumber(L, arg),buff,16);
```
Comment (RodADavison) [inline lib/lua/src/lstrlib.c:860]: should this be: modp_itoa16((LUA_INTFRM_T)luaL_checknumber(L, arg),buff); ?

Comment (RodADavison) [inline lib/lua/src/lstrlib.c:860]: It doesn't look like the new function uitoa16 is used/needed?!

Comment (eacmen) [reply lib/lua/src/lstrlib.c:860]: You are correct. I made this change a long time ago and forgot that the lua code already had the function I needed. Thanks for finding that!

**Comment by eacmen** [comment]
closing this PR and basing it off of `r/2.18.2` insteads.

**Pull request closed without merging**